docs/lib/index.rst

```diff
@@ -0,0 +1,221 @@
+Starlite library documentation
+==============================
+
+
+Starlite is a powerful, flexible, highly performant and opinionated ASGI framework,
+offering first class typing support and a full `Pydantic <https://github.com/samuelcolvin/pydantic>`_
+integration.
+
+The Starlite framework supports :doc:`/lib/usage/plugins/index`, ships
+with :doc:`dependency injection </lib/usage/dependency-injection>`, :doc:`security primitives </lib/usage/security/index>`,
+:doc:`OpenAPI schema generation </lib/usage/openapi>`, `MessagePack <https://msgpack.org/>`_,
+:doc:`middlewares </lib/usage/middleware/index>` and much more.
+
+Installation
+------------
+
+.. code-block:: shell
+
+   pip install starlite
+
+
+.. dropdown:: Extras
+    :icon: star
+
+    :ref:`Brotli Compression Middleware <lib/usage/middleware/builtin-middleware:brotli>`:
+        :code:`pip install starlite[brotli]`
+
+    :ref:`Client-side sessions <lib/usage/middleware/builtin-middleware:client-side sessions>`
+        :code:`pip install starlite[cryptography]`
+
+    :ref:`Server-side sessions <lib/usage/middleware/builtin-middleware:redis storage>` / :ref:`redis caching <lib/usage/caching:redis>`:
+        :code:`pip install starlite[redis]`
+
+    :ref:`Server-side sessions <lib/usage/middleware/builtin-middleware:memcached storage>` / :ref:`redis caching <lib/usage/caching:memcached>`:
+        :code:`pip install starlite[memcached]`
+
+    :ref:`Picologging <lib/usage/the-starlite-app:using picologging>`
+        :code:`pip install starlite[picologging]`
+
+    :ref:`StructLog <lib/usage/the-starlite-app:using structlog>`
+        :code:`pip install starlite[structlog]`
+
+    :doc:`/lib/usage/contrib/open-telemetry`
+        :code:`pip install starlite[openetelemetry]`
+
+    :doc:`/lib/usage/cli`
+        :code:`pip install starlite[cli]`
+
+    Standard installation (includes CLI, picologging and Jinja2 templating):
+        :code:`pip install starlite[standard]`
+
+    All extras:
+        :code:`pip install starlite[full]`
+
+
+
+Minimal Example
+---------------
+
+**Define your data model** using pydantic or any library based on it (for example ormar, beanie, SQLModel):
```

Comment:
We allow using dataclasses, typed dicts etc. I suggest we don't get bogged down with pydantic.

Reply:
Agree, but I don't want to change docs content in this PR. This should be purely structural.